server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java

```diff
@@ -97,13 +105,21 @@ public Message fromMetaDataWithContent(MetaDataWithContent message) throws Mailb
                 .replyTo(fromAddressList(mimeMessage.getReplyTo()))
                 .size(message.getSize())
                 .date(message.getInternalDateAsZonedDateTime())
-                .textBody(messageContent.getTextBody().orElse(null))
-                .htmlBody(messageContent.getHtmlBody().orElse(null))
+                .textBody(messageContent.getTextBody().orElse(textBody))
+                .htmlBody(htmlBody)
                 .preview(getPreview(messageContent))
```

Comment:
To compute the preview you can now rely on messagePreview.forTextBody() on the text body you just computed.

This will avoid doing two times HTML -> text convertion